Implemented Brave Rewards Ads feedback loop

jasonrsadler opened the pull request:
Partial for brave/brave-browser#4047
## Changes
Implemented Ads feedback loop

## Test plan
##### Link / storybook path to visual changes

- https://brave-ui-o7946mh5q.now.sh/?path=/story/feature-components-rewards-modal--show-ads-history
<!-- can be localhost storybook or `now` deployment -->

## Integration
- [ ] Does this contain changes to src/components or src/
  - [ ] Will you publish to npm immediately after this PR, or wait until sometime in the future?
  - [ ] Incompatible API change to something existing _(major version increase)_
  - [ ] Adding new backwards-compatible functionality? _(minor version increase)_
  - [ ] Fixing a bug backwards-compatibly? _(patch version increase)_
  
- [ ] Does this contain changes to src/features for brave-core?
  - [ ] Are there non backwards-compatible changes required for brave-core? **Do not merge until brave-core PR is approvable.** Link to brave-core PR:
  - [ ] Will you create brave-core PR to update to this commit after it is merged?
  - [ ] Wants uplift to brave-core feature branch?
     - When uplift-approved, merge to brave-core-0.VV.x feature branch
     - Create additional brave-core PRs for each feature branch to update commit

ryanml commented on package/package.json:
```diff
@@ -2,7 +2,7 @@
   "main": "index.js",
   "sideEffects": false,
-  "version": "0.38.0",
+  "version": "0.34.4",
```
We shouldn't need to make updates to this file anymore (unless we are doing an NPM publish)

Was this inadvertent or did `react-storybook-addon-chapters` need that bump?

jasonrsadler replied:
inadvertent for both

jasonrsadler replied:
Removed.

ryanml commented on src/features/rewards/tableAdsHistory/index.tsx:
```diff
@@ -0,0 +1,240 @@
+              this.getAdContent(row.adContent, i)
+            ),
+            customStyle: {
+              width: '70% !important',
```
I know we have these inline styles in the `tableContribute` component, but could we offload these to styled components? Let me know if there was some other limitation

jasonrsadler replied:
Offloaded. In fact, only the td was effective.

jasonrsadler commented:
Additional design req's:

1. Change 30 days to 7 days
2. Change dashboard link from 'Show ads history' to '7 day ads history'
3. Add thumb up filter
4. Add tooltip to opt out icon
5. Change box link to disabled rather than invisible
6. `Landed` status will show as 'Clicked' on UI

cc @jenn-rhim @alexwykoff

jasonrsadler commented:
Ready for review

emerick commented:
@jasonrsadler The backend will pass the `adAction` as a string that we should use as a localization key for looking up the locale-independent string.

Mapping would look like this, from what I gather though I'm only particular about the left sides:

`click => Clicked`
`dismiss => Closed`
`view => Viewed`
`landed => Clicked`

NejcZdovc commented on src/features/rewards/adRowsDetails/index.tsx:
```diff
@@ -0,0 +1,99 @@
+              <StyledAdsDateRow>
+                <StyledCaratIcon onClick={this.setInnerVisible}>
+                {
+                  this.state.innerDetailVisible ?
```
nit: format

```
this.state.innerDetailVisible 
? <CaratTriangleDownSIcon />
: <CaratTriangleRightSIcon />
```

jasonrsadler replied:
fixed

NejcZdovc commented on src/features/rewards/adRowsDetails/index.tsx:
```diff
@@ -0,0 +1,99 @@
+            <tr key={j}>
+            {
+              detailRow.content.map((detailCell: Cell, k: number) => {
+                return k === 0 ?
```
nit: format

jasonrsadler replied:
fixed

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+import styled from 'styled-components'
+
+export const StyledDateText = styled<{}, 'div'>('div')`
+  font-family: Muli, sans-serif;
```
please use theme font family

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+export const StyledDateText = styled<{}, 'div'>('div')`
+  font-family: Muli, sans-serif;
+  font-size: 14px;
+  color: #686978;
```
please use theme colors

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+`
+
+export const StyledHR = styled<{}, 'hr'>('hr')`
+  color: #DADCE8;
```
please use theme colors

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+
+export const StyledHR = styled<{}, 'hr'>('hr')`
+  color: #DADCE8;
+  background-color: #DADCE8;
```
please use theme colors

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+  cursor: pointer;
+  width: 16px;
+  height: 16px;
+  color: #9E9FAB;
```
please use theme colors

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+  font-family: Muli, sans-serif;
+  font-size: 14px;
+  font-weight: 500;
+  color: #686978;
```
please use theme colors

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+`
+
+export const StyledInnerStartTD = styled<{}, 'td'>('td')`
+  font-family: Muli, sans-serif;
```
please use theme font family

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+  font-family: Muli, sans-serif;
+  font-size: 14px;
+  font-weight: 500;
+  color: #686978;
```
please use theme colors

NejcZdovc commented:
I would say that left side should be one the same line as the right side

![image](https://user-images.githubusercontent.com/9574457/60869188-c8a59180-a22e-11e9-86b2-40f0e8db2192.png)

NejcZdovc commented:
Selection I would say is off

`All tab`
![image](https://user-images.githubusercontent.com/9574457/60869416-3487fa00-a22f-11e9-98a7-b6b4dd225daa.png)
All should be blue, Saved and Like should be gray

`Saved`
![image](https://user-images.githubusercontent.com/9574457/60869453-47023380-a22f-11e9-8afc-132f8b7ec522.png)
Saved should be blue, All and Like should be gray

`Like`
![image](https://user-images.githubusercontent.com/9574457/60869473-541f2280-a22f-11e9-8001-f0db564cd390.png)
Like should be red, All and Saved should be gray

NejcZdovc commented:
Whole row should be toggable not just carrot 

![image](https://user-images.githubusercontent.com/9574457/60869565-80d33a00-a22f-11e9-9ea3-d1463c46441a.png)

NejcZdovc commented:
If you move mouse down and click (you need to be fast and do it couple of times, it happened to me unintentionally at first) hover notification is stuck and you can't dismiss it (tried clicking out side and on it)

![image](https://user-images.githubusercontent.com/9574457/60869719-cabc2000-a22f-11e9-8028-ed3f0e694d7e.png)

NejcZdovc commented on src/features/rewards/categoryLikePicker/index.tsx:
```diff
@@ -0,0 +1,113 @@
+
+  render () {
+    return (
+      this.state.itemSelected === 1 ?
```
maybe moving this into separate function so that it's easier to read it

jasonrsadler replied:
Donel

NejcZdovc commented on src/features/rewards/adRowsDetails/style.ts:
```diff
@@ -0,0 +1,74 @@
+`
+
+export const StyledAdPortionTD = styled<{}, 'td'>('td')`
+  font-family: Muli, sans-serif;
```
please use theme font family

NejcZdovc commented on src/features/rewards/categoryLikePicker/style.ts:
```diff
@@ -0,0 +1,39 @@
+`
+
+export const StyledCategoryActionOptinFilledButton = styled(StyledCategoryActionOptinButton)`
+  color: red;
```
please use theme colors

NejcZdovc commented on src/features/rewards/categoryLikePicker/style.ts:
```diff
@@ -0,0 +1,39 @@
+`
+
+export const StyledCategoryActionOptoutFilledButton = styled(StyledCategoryActionOptoutButton)`
+  color: red;
```
please use theme colors

NejcZdovc commented on src/features/rewards/dropMenu/style.ts:
```diff
@@ -0,0 +1,46 @@
+  margin-top: -10px;
+  background-color: white;
+  border: 1px solid;
+  border-color: #DADCE8;
```
please use theme colors

NejcZdovc commented on src/features/rewards/dropMenu/style.ts:
```diff
@@ -0,0 +1,46 @@
+  margin-top: auto;
+  cursor: pointer;
+  &:hover {
+    background-color: #B0DBFF;
```
please use theme colors

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/index.tsx:
```diff
@@ -0,0 +1,170 @@
+  rows?: DetailRow[]
+  onClose?: () => void
+  id?: string
+  isMobile?: boolean
```
do we need this?

jasonrsadler replied:
eventually

NejcZdovc replied:
@emerick please remove

NejcZdovc commented:
Text are going over

![image](https://user-images.githubusercontent.com/9574457/60870823-e1637680-a231-11e9-9092-bd2aea42c933.png)

NejcZdovc commented:
We need to add padding around category text

![image](https://user-images.githubusercontent.com/9574457/60871005-356e5b00-a232-11e9-8d1c-a8da117ff04f.png)

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+import styled from 'styled-components'
+
+export const StyledWrapper = styled<{}, 'div'>('div')`
+  font-family: Poppins, sans-serif;
```
please use theme font

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+export const StyledLink = styled<{}, 'a'>('a')`
+  cursor: pointer;
+  display: inline-block;
+  color: #696FDC;
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+
+export const StyledDisabledLink = styled<{}, 'span'>('span')`
+  display: inline-block;
+  color: #CED0DB;
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+  font-size: 16px;
+  letter-spacing: 0;
+  display: inline-block;
+  color: #C2C4CF;
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+export const StyledAdsHistoryTitle = styled<{}, 'div'>('div')`
+  font-size: 22px;
+  font-weight: 600;
+  color: rgb(193, 45, 124);
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+
+export const StyledNote = styled<{}, 'div'>('div')`
+  max-width: 508px;
+  font-family: Muli,sans-serif;
```
please use theme font

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+  font-weight: 300;
+  margin-bottom: 10px;
+  line-height: 1.5;
+  color: #686978;
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+  font-weight: 200;
+  letter-spacing: 0;
+  display: inline-block;
+  color: #C2C4CF
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+
+export const StyledAdsInfoText = styled<{}, 'span'>('span')`
+  font-size: 18px;
+  color: rgb(193, 45, 124);
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,119 @@
+  display: inline-block;
+  width: 32px;
+  height: 32px;
+  color: #CED0DB;
```
please use theme color

NejcZdovc commented on src/features/rewards/showAdsHistory/style.ts:
```diff
@@ -0,0 +1,29 @@
+export const StyledLink = styled<{}, 'a'>('a')`
+  cursor: pointer;
+  display: inline-block;
+  color: #696FDC;
```
please use theme color

NejcZdovc commented on src/features/rewards/showAdsHistory/style.ts:
```diff
@@ -0,0 +1,29 @@
+
+export const StyledDisabledLink = styled<{}, 'span'>('span')`
+  display: inline-block;
+  color: #CED0DB;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/index.tsx:
```diff
@@ -0,0 +1,245 @@
+import { getLocale } from '../../../helpers'
+
+export interface DetailRow {
+  [key: string]: any
```
why do we need this one?

NejcZdovc commented on src/features/rewards/tableAdsHistory/index.tsx:
```diff
@@ -0,0 +1,245 @@
+  children?: React.ReactNode
+  headerColor?: boolean
+  rows?: DetailRow[]
+  allItems?: number
```
what this one means? if I understand correctly is what tab do we want to show?

jasonrsadler replied:
There are no tabs in this design.

This is used to display ad items according to the 'All' | 'Saved' | 'thumb-up' filter

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  width: 15%;
+`
+
+export const StyledTHFirst = styled<{}, 'th'>('th')`
```
what is the difference between `StyledTHFirst` and `StyledTHLast`. Also it looks like `StyledTHOther` and `StyledTHFirst` only have `width` different, so not sure why we duplicate everything else.

jasonrsadler replied:
Enough has changed to render these unnecessary. Removed, and replaced with single instance.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-family: ${p => p.theme.fontFamily.body};
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
+  color: #4B4C5C;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-family: ${p => p.theme.fontFamily.body};
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
+  color: #4B4C5C;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-family: ${p => p.theme.fontFamily.body};
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
```
please use theme color

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-size: 12px;
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  font-weight: 500;
+  border-bottom: 2px solid #dedfe4;
+  border-top: 2px solid #dedfe4;
+  color: #4B4C5C;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+export const StyledToggle = styled<{}, 'div'>('div')`
+  font-family: Poppins, sans-serif;
+  font-size: 13px;
+  color: #4c54d2;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+`
+
+export const StyledToggle = styled<{}, 'div'>('div')`
+  font-family: Poppins, sans-serif;
```
please use theme font

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  text-decoration: none;
+  cursor: pointer;
+  display: flex;
+  color: #686978
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  border: 1px solid;
+  border-collapse: separate;
+  border-radius: 5px;
+  border-color: #DADCE8;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+  height: 64px;
+`
+
+// TODO(jsadler): 'display: none' becomes 'display: inline-block' once icons are in place
```
is this TODO still needed?

jasonrsadler replied:
no

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+`
+export const StyledAdInfo = styled<{}, 'div'>('div')`
+  display: block;
+  color: #AEB1C2;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+`
+
+export const StyledAdBrand = styled<{}, 'div'>('div')`
+  font-family: Poppins, sans-serif;
```
please use theme font

emerick replied:
Fixed.

NejcZdovc commented on src/features/rewards/tableAdsHistory/style.ts:
```diff
@@ -0,0 +1,171 @@
+`
+
+export const StyledTD = styled<{}, 'td'>('td')`
+  width: '15%';
```
`'15%'` -> `15%`

jasonrsadler replied:
Fixed

NejcZdovc commented on src/features/rewards/thumbLikePicker/style.ts:
```diff
@@ -0,0 +1,30 @@
+`
+
+export const StyledAdStatThumbUpFilledIcon = styled(StyledAdStatThumbUpIcon)`
+  color: red;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc replied:
not seeing the change, could you please check again

NejcZdovc commented on src/features/rewards/thumbLikePicker/style.ts:
```diff
@@ -0,0 +1,30 @@
+`
+
+export const StyledAdStatThumbDownFilledIcon = styled(StyledAdStatThumbDownIcon)`
+  color: red;
```
please use theme color

emerick replied:
Fixed.

NejcZdovc replied:
not seeing the change, could you please check again

NejcZdovc commented:
When there is no ads text I think should be gray like it is in other places. Right now it looks like paragraph

![image](https://user-images.githubusercontent.com/9574457/60872193-8ed78980-a234-11e9-9e5e-92bf9618f61b.png)


What we have in other places
![image](https://user-images.githubusercontent.com/9574457/60872246-a31b8680-a234-11e9-893d-3bddb64b974e.png)

jasonrsadler commented:
> I would say that left side should be one the same line as the right side
> 
> ![image](https://user-images.githubusercontent.com/9574457/60869188-c8a59180-a22e-11e9-86b2-40f0e8db2192.png)

Fixed.


> Selection I would say is off
> 
> `All tab`
> ![image](https://user-images.githubusercontent.com/9574457/60869416-3487fa00-a22f-11e9-98a7-b6b4dd225daa.png)
> All should be blue, Saved and Like should be gray
> 
> `Saved`
> ![image](https://user-images.githubusercontent.com/9574457/60869453-47023380-a22f-11e9-8afc-132f8b7ec522.png)
> Saved should be blue, All and Like should be gray
> 
> `Like`
> ![image](https://user-images.githubusercontent.com/9574457/60869473-541f2280-a22f-11e9-8001-f0db564cd390.png)
> Like should be red, All and Saved should be gray

No. Gray indicates it is not clickable because the user already has it selected.



> Whole row should be toggable not just carrot
> 
> ![image](https://user-images.githubusercontent.com/9574457/60869565-80d33a00-a22f-11e9-9ea3-d1463c46441a.png)

Done.



> If you move mouse down and click (you need to be fast and do it couple of times, it happened to me unintentionally at first) hover notification is stuck and you can't dismiss it (tried clicking out side and on it)
> 
> ![image](https://user-images.githubusercontent.com/9574457/60869719-cabc2000-a22f-11e9-8028-ed3f0e694d7e.png)

I cannot reproduce this Bx

> Text is going over
> 
> ![image](https://user-images.githubusercontent.com/9574457/60870823-e1637680-a231-11e9-9092-bd2aea42c933.png)

Fixed.



> We need to add padding around category text
> 
> ![image](https://user-images.githubusercontent.com/9574457/60871005-356e5b00-a232-11e9-8d1c-a8da117ff04f.png)

Fixed

> When there is no ads text I think should be gray like it is in other places. Right now it looks like paragraph
> 
> ![image](https://user-images.githubusercontent.com/9574457/60872193-8ed78980-a234-11e9-9e5e-92bf9618f61b.png)
> 
> What we have in other places
> ![image](https://user-images.githubusercontent.com/9574457/60872246-a31b8680-a234-11e9-893d-3bddb64b974e.png)

This design was implemented in collaboration with @jenn-rhim and @alexwykoff and will defer to them on current design implementation.

For theme color and font comments, there currently is no 'Rewards' theme and the colors and fonts used are done so to match the spec. cc @rossmoody on 'Rewards' theme concept.

jenn-rhim commented:
@NejcZdovc Let's not display the Ads History link or disable the link when the user did not opt in to ads.

jenn-rhim commented:
@jasonrsadler For 'You're currently receiving...', for narrower window, it'd make it difficult to keep the filter in the same line while not running into each other. We should consider shortening it to 'Your setting: maximum 2 ads per hour'.

jasonrsadler commented:
Done.
<img width="882" alt="Screen Shot 2019-07-12 at 2 05 10 PM" src="https://user-images.githubusercontent.com/25326067/61149025-25fb4600-a4ae-11e9-9253-e74afe0c9cda.png">

jasonrsadler commented:
cc @jenn-rhim 
Filters updated

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/index.tsx:
```diff
@@ -0,0 +1,170 @@
+  id?: string
+  isMobile?: boolean
+  adsPerHour?: number
+  adsPerDay?: number
```
let's improve naming on this one. This one is used to display `Ads you've received in the past 7 days`.  Maybe `totalDays`?

tmancey replied:
adsPerDay is correct which matches bat-native-ads?

NejcZdovc replied:
this one is not the same

NejcZdovc replied:
![image](https://user-images.githubusercontent.com/9574457/62478420-5bdcd300-b7ab-11e9-8abe-eb6a4645c871.png)

this variable is for this title

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/index.tsx:
```diff
@@ -0,0 +1,170 @@
+            {getLocale('adsHistoryTitle')}
+          </StyledAdsHistoryTitle>
+          <StyledSubTitleText>
+            {getLocale('adsHistorySubTitle')}
```
this one needs to accept variable as currently is fixed to `7 Days`. We need to pass in `totalDays`

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/index.tsx:
```diff
@@ -0,0 +1,170 @@
+                {getLocale('adsCurrentlyViewing')}
+              </StyledAdsInfoText>
+              <StyledAdsPerHourText>
+                {(adsPerHour || '0') + (adsPerHour !== 1 ? getLocale('adsPerHourPlural') : getLocale('adsPerHourSingular'))}
```
this will not work for all languages. In Slovenian we have special variation for double as well

NejcZdovc commented on src/features/rewards/tableAdsHistory/index.tsx:
```diff
@@ -0,0 +1,244 @@
+  brandUrl: string
+  brandDisplayUrl: string
+  likeAction: number
+  adAction: 'click' | 'dismiss' | 'view' | 'landed'
```
let's export this as it could be useful for client when doing checks

NejcZdovc commented:
@emerick please check my previous comments about theme color and fonts. Thank you

emerick commented:
@NejcZdovc Think I addressed all of the open feedback items, please take a look when you have a moment. Thanks!

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,118 @@
+
+export const StyledNotSelectedLink = styled<{}, 'span'>('span')`
+  display: inline-block;
+  color: #696FDC;
```
please use theme color

NejcZdovc commented on src/features/rewards/modalShowAdsHistory/style.ts:
```diff
@@ -0,0 +1,118 @@
+export const StyledLink = styled<{}, 'a'>('a')`
+  cursor: pointer;
+  display: inline-block;
+  color: #CED0DB;
```
please use theme color
